sdk/py/apepay/manager.py

```diff
@@ -87,132 +91,131 @@ def _parse_validator(self) -> Callable[[_ValidatorItem], Validator]:
     def set_validators(self) -> ContractTransactionHandler:
 
         @wraps(self.contract.set_validators)
-        def order_validators(*validators: _ValidatorItem, **txn_kwargs) -> ReceiptAPI:
+        def set_validators(*validators: _ValidatorItem, **txn_kwargs) -> ReceiptAPI:
             # NOTE: Always keep sets sorted, ensure no duplicates
-            return self.contract.set_validators(
-                sorted(v.address for v in set(map(self._parse_validator, validators))),
-                **txn_kwargs,
-            )
+            new_validators = sorted(v.address for v in set(map(self._parse_validator, validators)))
+            # NOTE: Okay to use Click here, intended primarily to support interactive features
+            click.echo("Validators update:")
+            for line in Differ().compare(tuple(v.address for v in self.validators), new_validators):
+                click.echo(line)
+            return self.contract.set_validators(new_validators, **txn_kwargs)
 
-        return cast(ContractTransactionHandler, order_validators)
+        return cast(ContractTransactionHandler, set_validators)
 
     def add_validators(self, *new_validators: _ValidatorItem, **txn_kwargs) -> ReceiptAPI:
         return self.set_validators(
             *(set(self.validators) | set(map(self._parse_validator, new_validators))),
             **txn_kwargs,
         )
 
+    def replace_validator(
+        self,
+        old_validator: _ValidatorItem,
+        new_validator: _ValidatorItem,
+        **txn_kwargs,
+    ) -> ReceiptAPI:
+        return self.set_validators(
+            *(
+                (set(self.validators) - set([self._parse_validator(old_validator)]))
+                | set([self._parse_validator(new_validator)])
+            ),
+            **txn_kwargs,
+        )
+
     def remove_validators(self, *old_validators: _ValidatorItem, **txn_kwargs) -> ReceiptAPI:
         return self.set_validators(
             *(set(self.validators) - set(map(self._parse_validator, old_validators))),
             **txn_kwargs,
         )
 
-    @property
-    def add_token(self) -> ContractTransactionHandler:
-        return self.contract.add_token
+    def add_token(self, token: AddressType, **txn_kwargs) -> ReceiptAPI:
+        return self.contract.set_token_accepted(token, True, **txn_kwargs)
 
-    @property
-    def remove_token(self) -> ContractTransactionHandler:
-        return self.contract.remove_token
+    def remove_token(self, token: AddressType, **txn_kwargs) -> ReceiptAPI:
+        return self.contract.set_token_accepted(token, False, **txn_kwargs)
 
-    @property
-    def is_accepted(self) -> ContractCallHandler:
-        return self.contract.token_is_accepted
+    def is_accepted(self, token: AddressType) -> bool:
+        return self.contract.token_is_accepted(token)
```

Comment:
We could potentially use the ape-extras mechanism and feed into `.contract` so all pass throughs are automatic.

Reply:
How does that work again?